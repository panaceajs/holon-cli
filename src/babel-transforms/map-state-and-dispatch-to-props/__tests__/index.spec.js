const pluginTester = require('babel-plugin-tester');
const mapStateAndDispatchToPropsTest = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const pluginJsx = require('@babel/plugin-syntax-jsx');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: mapStateAndDispatchToPropsTest,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginJsx, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should add nothing when there is nothing to add.',
      fixture: '__fixtures__/bare-connect.js',
      snapshot: false
    },
    {
      title: 'it should add map state to props.',
      fixture: '__fixtures__/bare-connect.js',
      pluginOptions: {
        mapProps: ['first']
      }
    },
    {
      title: 'it should add map dispatch to props.',
      fixture: '__fixtures__/bare-connect.js',
      pluginOptions: {
        actionTypes: { firstAction: 'first', secondAction: 'second' }
      }
    },
    {
      title: 'it should leverage stateNamespace.',
      fixture: '__fixtures__/bare-connect.js',
      pluginOptions: {
        stateNamespace: 'namespace',
        actionTypes: {
          firstAction: 'first',
          secondAction: 'second'
        }
      }
    }
  ]
});
