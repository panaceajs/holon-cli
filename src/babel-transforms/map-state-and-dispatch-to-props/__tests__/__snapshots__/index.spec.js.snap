// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-map-state-and-dispatch-to-props it should add map dispatch to props.: it should add map dispatch to props. 1`] = `
"
import { connect } from 'react-redux';
const C = () => <div></div>;
export default connect()(C);

      ↓ ↓ ↓ ↓ ↓ ↓

import { connect } from 'react-redux';

const mapStateToProps = ({
  first,
  second
}) => ({
  first: first,
  second: second
});

const mapDispatchToProps = dispatch => ({
  firstAction: payload => {
    dispatch(firstAction(payload));
  },
  secondAction: payload => {
    dispatch(secondAction(payload));
  }
});

const C = () => <div></div>;
export default connect(mapStateToProps, mapDispatchToProps)(C);
"
`;

exports[`holon-map-state-and-dispatch-to-props it should add map state to props.: it should add map state to props. 1`] = `
"
import { connect } from 'react-redux';
const C = () => <div></div>;
export default connect()(C);

      ↓ ↓ ↓ ↓ ↓ ↓

import { connect } from 'react-redux';

const mapStateToProps = ({
  first
}) => ({
  first: first
});

const C = () => <div></div>;
export default connect(mapStateToProps, null)(C);
"
`;

exports[`holon-map-state-and-dispatch-to-props it should leverage stateNamespace.: it should leverage stateNamespace. 1`] = `
"
import { connect } from 'react-redux';
const C = () => <div></div>;
export default connect()(C);

      ↓ ↓ ↓ ↓ ↓ ↓

import { connect } from 'react-redux';

const mapStateToProps = ({
  namespace: {
    first,
    second
  }
}) => ({
  first: first,
  second: second
});

const mapDispatchToProps = dispatch => ({
  firstAction: payload => {
    dispatch(firstAction(payload));
  },
  secondAction: payload => {
    dispatch(secondAction(payload));
  }
});

const C = () => <div></div>;
export default connect(mapStateToProps, mapDispatchToProps)(C);
"
`;
